src/Shared/NativeMethodsShared.cs

```diff
@@ -613,17 +613,42 @@ internal static string FrameworkBasePath
         /// <summary>
         /// System information, initialized when required.
         /// </summary>
-        private static readonly Lazy<SystemInformationData> SystemInformation = new Lazy<SystemInformationData>(true);
+        /// <remarks>
+        /// Initially implemented as <see cref="Lazy{SystemInformationData}"/>, but
+        /// that's .NET 4+, and this is used in MSBuildTaskHost.
+        /// </remarks>
+        private static SystemInformationData SystemInformation
+        {
+            get
+            {
+                if (!_systemInformationInitialized)
+                {
+                    lock (SystemInformationLock)
+                    {
+                        if (!_systemInformationInitialized)
+                        {
+                            _systemInformation = new SystemInformationData();
+                            _systemInformationInitialized = true;
+                        }
+                    }
+                }
+                return _systemInformation;
+            }
+        }
+
+        private static SystemInformationData _systemInformation;
+        private static bool _systemInformationInitialized;
```

Comment:
[Allegedly](http://stackoverflow.com/a/1964832) unnecessary--if two threads get stale cached values, they'll both attempt to acquire the lock, one will succeed and pass the inner check, the other will block, then get the lock and fail the inner check.